Find timezone on Solaris without having systemd's timedated

Will not be accepted upstream, since they refuse to support any API other
than timedated, so we'll need to provide timedated at some point.

--- a/panels/datetime/cc-datetime-panel.c	2019-11-15 20:19:49.735182510 -0800
+++ b/panels/datetime/cc-datetime-panel.c	2019-11-15 20:21:58.362476356 -0800
@@ -586,7 +586,29 @@
 {
   const gchar *timezone;
 
+#ifdef __sun // no support for systemd's timedated yet
+  timezone = getenv("TZ");
+  if (timezone == NULL || (strcmp(timezone, "localtime") == 0))
+    {
+      char linkbuf[PATH_MAX];
+      ssize_t bufused;
+      const char tzdir[] = "/usr/share/lib/zoneinfo/";
+
+      bufused = readlink("/etc/localtime", linkbuf, sizeof(linkbuf) - 1);
+      if (bufused > (sizeof(tzdir) - 1))
+      {
+        char *p;
+        linkbuf[bufused] = '\0';
+        p = strstr(linkbuf, tzdir);
+        if (p != NULL)
+        {
+          timezone = g_strdup(p + sizeof(tzdir) - 1);
+        }
+      }
+    }
+#else // use systemd
   timezone = timedate1_get_timezone (self->dtm);
+#endif
 
   if (timezone == NULL ||
       !cc_timezone_map_set_timezone (CC_TIMEZONE_MAP (self->map), timezone))
